### PR TITLE
Add label to LightningKokkos GPU CIs

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** Add label to LightningKokkos GPU CIs
Also fixed the workflow activation logic.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-74149]

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,9 @@
 
 ### Improvements
 
+* Add label to `lightning.kokkos` GPU CIs.
```
Comment (maliasadi) [inline .github/CHANGELOG.md:37]:
```suggestion
* Add the `ci:use-gpu-runner` GitHub label to `lightning.kokkos` GPU Testing CIs.
```

Comment (AmintorDusko) [reply .github/CHANGELOG.md:37]: Thanks!

**Review by maliasadi — APPROVED** [review]
Thanks @AmintorDusko!

**Review by multiphaseCFD — APPROVED** [review]
Thanks @AmintorDusko for the nice work!

**Review by maliasadi — APPROVED** [review]
:partying_face:

**Pull request merged**